packages/flutter_tools/lib/src/observatory.dart

```diff
@@ -113,10 +115,17 @@ class Observatory {
     });
   }
 
-  Future<Response> isolateReload(String isolateId) {
-    return sendRequest('isolateReload', <String, dynamic>{
-      'isolateId': isolateId
-    });
+  Future<bool> reloadSources(String isolateId) async {
+    Future<Event> future = onIsolateEvent
+      .where((Event event) => event.kind == 'IsolateReload')
+      .single;
+
+    await sendRequest('_reloadSources', <String, dynamic>{ 'isolateId': isolateId });
+    Event event = await future.timeout(new Duration(seconds: 20));
```

Comment:
It's an infrequently used broadcast stream, and I think it's ok to leak it here. I looked at re-writing it. There's no access to the subscription used by `single`; I'd need to add a bunch of logic to to what `single` does in a way where I could cancel the subscription.